services-api/src/main/java/io/scalecube/services/Reflect.java

```diff
@@ -269,8 +275,22 @@ public static void validateMethodOrThrow(Method method) {
     } else if (!Publisher.class.isAssignableFrom(returnType)) {
       throw new UnsupportedOperationException("Service method return type can be Publisher only");
     }
-    if (method.getParameterCount() > 1) {
-      throw new UnsupportedOperationException("Service method can accept 0 or 1 parameters only");
+    if (method.getParameterCount() == 2) {
```

Comment:
this block can be improved... 
maybe consider to make verification method  since reading all these if is off context to the logical flow
something like
verifyPrincipelParam ...????

Reply:
Fixed in https://github.com/scalecube/scalecube-services/pull/624/commits/824e427df1b98fe1640efbfe321e1471c706e37f